packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter.tsx

```diff
@@ -7,23 +7,25 @@ import { SynapseConstants } from '../../../utils'
 import useGetInfoFromIds from '../../../utils/hooks/useGetInfoFromIds'
 import {
   ColumnTypeEnum,
+  EntityHeader,
   Evaluation,
-  UserGroupHeader,
-} from '@sage-bionetworks/synapse-types'
-import { EntityHeader } from '@sage-bionetworks/synapse-types'
-import {
   FacetColumnResultValueCount,
+  FacetColumnValuesRequest,
+  QueryBundleRequest,
   SelectColumn,
+  UserGroupHeader,
 } from '@sage-bionetworks/synapse-types'
 import IconSvg from '../../IconSvg/IconSvg'
 import { Checkbox } from '../Checkbox'
 import { FacetFilterHeader } from './FacetFilterHeader'
+import { useQueryContext } from '../../QueryContext'
+import { ReadonlyDeep } from 'type-fest'
+import { isFacetColumnValuesRequest } from '../../../utils/types/IsType'
+import { cloneDeep } from 'lodash-es'
 
 export type EnumFacetFilterProps = {
   facetValues: FacetColumnResultValueCount[]
   columnModel: SelectColumn
-  onChange: (facetNamesMap: Record<string, boolean>) => void
-  onClear: () => void
```

Comment:
Now use the query context instead of these methods.

Ideally I would like to split this into a "View" component that has these props and a wrapper component that passes business logic into the View components' props, but I held off since that would be an even larger change.